### PR TITLE
CREATE_SPARTA_HANDLER const methods

### PR DESCRIPTION
**Pull request by avinabadasgupta**
#68

### CODE REVIEW

File: sparta/sparta/kernel/SpartaHandler.hpp [hunk 1]
```diff
@@ -251,6 +316,45 @@ namespace sparta
...
+     *     }
+     *
+     *     void myMethod() {
+     *         std::cout << "myMethod called" << std::endl;
+     *     }
+     *
+     * };
+     * \endcode
+     */
+#define CREATE_CONST_SPARTA_HANDLER(clname, meth)                            \
```
Comment (klingaard) [inline sparta/sparta/kernel/SpartaHandler.hpp]: Did you really need a separate macro?

Comment (avinabadasgupta) [reply sparta/sparta/kernel/SpartaHandler.hpp]: No. Got confused there for a bit. Changing.

Comment (avinabadasgupta) [reply sparta/sparta/kernel/SpartaHandler.hpp]: My latest commit is not really what you wanted but it's a start.
I removed the need for Sparta Handlers from registerConsumerHandler method of Ports.
Tested this on core example.

**Comment by klingaard** [comment]
So, this is a pretty big change.  Not sure if it will fly with folks.  But, I'm gonna challenge you once more...  I'll bet you can figure out the argument type of the function as well as determine the address of the function handed.  So you can go from this:
```
in_mss_ack_sync_.registerConsumerHandler<BIU, bool, &BIU::getAckFromMSS_>(this);
```
to this:
```
in_mss_ack_sync_.registerConsumerHandler(this, getAckFromMSS_);
```
After all, weren't you able to figure this out with your `PairDefinitions`?  :)

**Comment by avinabadasgupta** [comment]
> So, this is a pretty big change. Not sure if it will fly with folks. But, I'm gonna challenge you once more... I'll bet you can figure out the argument type of the function as well as determine the address of the function handed. So you can go from this:
> 
> ```
> in_mss_ack_sync_.registerConsumerHandler<BIU, bool, &BIU::getAckFromMSS_>(this);
> ```
> 
> to this:
> 
> ```
> in_mss_ack_sync_.registerConsumerHandler(this, &BIU::getAckFromMSS_);
> ```
> 
> After all, weren't you able to figure this out with your `PairDefinitions`? :)
> 
> BTW, the Port knows the expected type for the argument. There's a hint there.

Let me play with this for a little more.

**Pull request closed without merging**